src/backend/src/health/health.controller.ts

```diff
@@ -0,0 +1,23 @@
+
+import { Controller, Get } from '@nestjs/common';
+import {
+  HealthCheckService,
+  HttpHealthIndicator,
+  HealthCheck,
+} from '@nestjs/terminus';
+
+@Controller('health')
+export class HealthController {
+  constructor(
+    private health: HealthCheckService,
+    private http: HttpHealthIndicator,
+  ) {}
+
+  @Get()
+  @HealthCheck()
+  check() {
+    return this.health.check([
+      () => this.http.pingCheck('dpia-web', 'http://localhost:3000/api'),
```

Comment:
Sorry, I might be new on this `Terminus` usage. But looks like from the documentation from [here](https://docs.nestjs.com/recipes/terminus) - the purpose of Terminus is to do health/ping checks on an external site. Aren't we here pinging our own servers continuously consuming resources and memory?  I think this is something openshift should take care and not every deployed node instance.

Please let me know if my understanding is different. Happy to connect over call.

isn't the purpose of pingCheck /

Reply:
Then I think we can delete this package, just do a simple http check for hc endpoint

Reply:
yea all we need to do is expose a healthcheck endpoint to serve back a 200 status. Openshift deployment configs allow for the addition of readiness and liveness probes that will target this endpoint to determine whether or not the container is healthy.